

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Creating an Annotated Expression Matrix &mdash; gsforge v0.6beta documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/togglebutton.css" type="text/css" />
  <link rel="stylesheet" href="../_static/mystnb.css" type="text/css" />
  <link rel="stylesheet" href="../_static/custom.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/togglebutton.js"></script>
        <script>var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Creating Feature Sets and Collections" href="set_and_collection_creation.html" />
    <link rel="prev" title="Selecting and Comparing Genes with GSForge" href="tour.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> gsforge
          

          
          </a>

          
            
            
              <div class="version">
                0.6
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption" role="heading"><span class="caption-text">User Guide / Tutorial:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../index.html"> Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="tour.html"> Tour</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"> Annotated Expression Matrix Creation</a></li>
<li class="toctree-l1"><a class="reference internal" href="set_and_collection_creation.html"> Set and Collection Creation</a></li>
<li class="toctree-l1"><a class="reference internal" href="Interface_Guide.html"> Interface Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="Plotting_Guide.html"> Plotting Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="R_integration.html"> R Integration Guide</a></li>
<li class="toctree-l1"><a class="reference internal" href="analysis_resources.html"> Analysis Resources</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Reference Examples:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../reference_examples/plot_gallery/ref_plots.html"> Reference Plots</a></li>
<li class="toctree-l1"><a class="reference internal" href="../reference_examples/Normalizations/norm_index.html"> Normalization Examples</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Walkthroughs:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../walkthroughs/oryza_sativa/oryza_sativa_index.html"> Oryza sativa</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">API Reference:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../API/modules.html">API</a></li>
<li class="toctree-l1"><a class="reference internal" href="../development.html">Development</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">gsforge</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
      <li>Creating an Annotated Expression Matrix</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/user_guide/gem_creation.md.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="creating-an-annotated-expression-matrix">
<h1>Creating an Annotated Expression Matrix<a class="headerlink" href="#creating-an-annotated-expression-matrix" title="Permalink to this headline">¶</a></h1>
<p>An <code class="docutils literal notranslate"><span class="pre">AnnotatedGEM</span></code> is one of the three core data structures provided by <code class="docutils literal notranslate"><span class="pre">GSForge</span></code>, it contains:</p>
<ul class="simple">
<li><p>a gene expression matrices (GEM), with named coordinates for genes and samples.</p></li>
<li><p>sample annotations / descriptors.</p></li>
<li><p>gene annotation / descriptors.</p></li>
</ul>
<p>More than one transform of the GEM can be stored, so long as it shares the same coordinates.</p>
<p>Here we demonstrate the most common ways to load expression data into <code class="docutils literal notranslate"><span class="pre">GSForge</span></code>.</p>
<p><em><strong>Notebook setup</strong></em></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="kn">from</span> <span class="nn">os</span> <span class="kn">import</span> <span class="n">environ</span>
<span class="kn">from</span> <span class="nn">pathlib</span> <span class="kn">import</span> <span class="n">Path</span>
<span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
<span class="kn">import</span> <span class="nn">GSForge</span> <span class="k">as</span> <span class="nn">gsf</span>

<span class="n">OSF_PATH</span> <span class="o">=</span> <span class="n">Path</span><span class="p">(</span><span class="n">environ</span><span class="o">.</span><span class="n">get</span><span class="p">(</span><span class="s2">&quot;GSFORGE_DEMO_DATA&quot;</span><span class="p">,</span> <span class="n">default</span><span class="o">=</span><span class="s2">&quot;~/GSForge_demo_data/&quot;</span><span class="p">))</span><span class="o">.</span><span class="n">expanduser</span><span class="p">()</span><span class="o">.</span><span class="n">joinpath</span><span class="p">(</span><span class="s2">&quot;osfstorage&quot;</span><span class="p">,</span> <span class="s2">&quot;oryza_sativa&quot;</span><span class="p">)</span>
<span class="n">COUNT_PATH</span> <span class="o">=</span> <span class="n">OSF_PATH</span><span class="o">.</span><span class="n">joinpath</span><span class="p">(</span><span class="s2">&quot;GEMmakerGEMs&quot;</span><span class="p">,</span> <span class="s2">&quot;rice_heat_drought.GEM.raw.txt&quot;</span><span class="p">)</span>
<span class="n">LABEL_PATH</span> <span class="o">=</span> <span class="n">OSF_PATH</span><span class="o">.</span><span class="n">joinpath</span><span class="p">(</span><span class="s2">&quot;GEMmakerGEMs&quot;</span><span class="p">,</span> <span class="s2">&quot;raw_annotation_data&quot;</span><span class="p">,</span> <span class="s2">&quot;PRJNA301554.hydroponic.annotations.txt&quot;</span><span class="p">)</span>

<span class="c1"># Output path.</span>
<span class="n">GEM_PATH</span> <span class="o">=</span> <span class="n">OSF_PATH</span><span class="o">.</span><span class="n">joinpath</span><span class="p">(</span><span class="s2">&quot;AnnotatedGEMs&quot;</span><span class="p">,</span> <span class="s2">&quot;oryza_sativa_hisat2_raw.nc&quot;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="from-text-files">
<h2>From Text Files<a class="headerlink" href="#from-text-files" title="Permalink to this headline">¶</a></h2>
<p>If your count and annotation files have matching sample indices you can create an AnnotatedGEM in a single step:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">agem</span> <span class="o">=</span> <span class="n">gsf</span><span class="o">.</span><span class="n">AnnotatedGEM</span><span class="o">.</span><span class="n">from_files</span><span class="p">(</span>
    <span class="n">count_path</span><span class="o">=</span><span class="n">COUNT_PATH</span><span class="p">,</span>
    <span class="n">label_path</span><span class="o">=</span><span class="n">LABEL_PATH</span><span class="p">,</span>
    <span class="c1"># These are the default arguments passed to from_files,</span>
    <span class="c1"># to the individual calls to `pandas.read_csv`.</span>
    <span class="n">count_kwargs</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">),</span>
    <span class="n">label_kwargs</span><span class="o">=</span><span class="nb">dict</span><span class="p">(</span><span class="n">index_col</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">),</span>
<span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>It is not uncommon to have to wrangle sample or gene names to some degree.
Once complete you may supply the data as a pair of <code class="docutils literal notranslate"><span class="pre">pandas.Dataframe</span></code> or a single <code class="docutils literal notranslate"><span class="pre">xarray.DataSet</span></code> object.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">count_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">COUNT_PATH</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">0</span><span class="p">)</span>
<span class="c1"># Wrangle data here...</span>

<span class="n">label_df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">LABEL_PATH</span><span class="p">,</span> <span class="n">index_col</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">)</span>
<span class="n">label_df</span><span class="p">[</span><span class="s1">&#39;genotype&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">label_df</span><span class="p">[</span><span class="s1">&#39;genotype&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot; &quot;</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">]</span>
<span class="n">label_df</span><span class="p">[</span><span class="s1">&#39;time&#39;</span><span class="p">]</span> <span class="o">=</span> <span class="n">label_df</span><span class="p">[</span><span class="s1">&#39;time&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">str</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">expand</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="mi">0</span><span class="p">]</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span>
<span class="c1"># Perhaps even more wrangling...</span>

<span class="c1"># Then provide them to GSForge:</span>
<span class="k">del</span> <span class="n">agem</span>
<span class="n">agem</span> <span class="o">=</span> <span class="n">gsf</span><span class="o">.</span><span class="n">AnnotatedGEM</span><span class="o">.</span><span class="n">from_pandas</span><span class="p">(</span><span class="n">count_df</span><span class="o">=</span><span class="n">count_df</span><span class="p">,</span> <span class="n">label_df</span><span class="o">=</span><span class="n">label_df</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;Oryza sativa&quot;</span><span class="p">)</span>

<span class="k">if</span> <span class="ow">not</span> <span class="n">GEM_PATH</span><span class="o">.</span><span class="n">exists</span><span class="p">():</span>
    <span class="n">agem</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="n">GEM_PATH</span><span class="p">)</span>
    
<span class="n">agem</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_plain highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>&lt;GSForge.AnnotatedGEM&gt;
Name: Oryza sativa
Selected GEM Variable: &#39;counts&#39;
    Gene   55986
    Sample 475
</pre></div>
</div>
</div>
</div>
</section>
</section>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="set_and_collection_creation.html" class="btn btn-neutral float-right" title="Creating Feature Sets and Collections" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="tour.html" class="btn btn-neutral float-left" title="Selecting and Comparing Genes with GSForge" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2021, Tyler Biggs, Stephen Ficklin.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>